### ISSUE TITLE
Add mimetype icons to project window

### ISSUE THREAD

**Issue by diversys** [HaikuArchives/Paladin#122]
It would also be nice to see icons to the left of filenames. Without them current list is a bit bleak.

![image](https://user-images.githubusercontent.com/1316513/46081685-cc878c00-c1a5-11e8-9bbf-50883195abe5.png)

Also show checkmark icon to the right to represent if the file was already compiled or not.

![image](https://user-images.githubusercontent.com/1316513/46082052-c2b25880-c1a6-11e8-9bb9-5437282100f6.png)

**Comment by adamfowleruk** [comment]
Yes good ideas.

**Comment by ghost** [comment]
Good day,

Agreed on this one. I usally code on VSCode, and use the extension VSCodeIcons which makes things visually easier.

Therefore, besides the Icons, it would be nice that Paladin would be able to support Plugins, instead of being a monolithic work, I'm thinking of the VSCode marketplace.

Regards,
RR

**Comment by adamfowleruk** [comment]
Yes indeed. It already supports tools plug ins, but the api isn't documented. Language plugins are the obvious next step, Compiler plug ins, etc

**Comment by adamfowleruk** [comment]
Also FYI my own main development ide is also vscode
